src/bundle.c

```diff
@@ -2502,6 +2502,50 @@ gboolean load_manifest_from_bundle(RaucBundle *bundle, RaucManifest **manifest,
 	return res;
 }
 
+static gboolean read_complete_dm_device(gchar *dev, GError **error)
+{
+	int fd = -1;
+	g_autofree void* buf = NULL;
+	ssize_t r;
+	gboolean ret = TRUE;
+
+	g_return_val_if_fail(dev != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	fd = g_open(dev, O_RDONLY | O_CLOEXEC, 0);
+	if (fd < 0) {
+		int err = errno;
+		g_set_error(error,
+				G_FILE_ERROR,
+				g_file_error_from_errno(err),
+				"Failed to open %s: %s", dev, g_strerror(err));
+		return FALSE;
+	}
+
+	lseek(fd, 0, SEEK_SET);
+
+	buf = g_malloc0(65536);
+
+	for (guint sector = 0;; sector++) {
+		r = pread(fd, buf, sizeof(buf), sector*sizeof(buf));
+		if (r == 0)
+			break;
+		else if (r < 0)	{
+			int err = errno;
+			g_set_error(error,
+					G_FILE_ERROR,
+					g_file_error_from_errno(err),
+					"Check whole %s device failed: %s", dev, g_strerror(err));
```

Comment:
This could print the failing offset, to help with debugging. See `G_GOFFSET_FORMAT` for formatting these values.

Reply:
Now prints range within the corruption occured